python/GafferArnoldUI/ArnoldLightUI.py

```diff
@@ -34,20 +34,35 @@
 #
 ##########################################################################
 
+import functools
+
 import Gaffer
 import GafferArnold
 
-## \todo Refactor the GafferScene::Light base class so this can be
-# registered there, and work for all subclasses. The main issue is that
-# there is no simple generic way of querying the required "ai:light:"
-# prefix from the subclass.
-def __parameterUserDefault( plug ) :
+# Defer parameter metadata lookups to the internal shader node
+# aside from nodule:type, which we handle manually to prevent
+# some connectable plugs appearing.
+
+def __parameterMetadata( plug, key ) :
+
+	node = plug.node()
+	return Gaffer.Metadata.value( node["__shader"]["parameters"].descendant( plug.relativeName( node["parameters"] ) ), key )
+
+for key in [
+	"description",
+	"label",
+	"noduleLayout:label",
+	"layout:divider",
+	"layout:section",
+	"presetNames",
+	"presetValues",
+	"plugValueWidget:type",
+	"noduleLayout:visible",
+	"noduleLayout:label",
```

Comment:
Repeat of line 54. Actually, I'm not sure we need any of this (apart from the original user default code), since ArnoldLightUI is already making all these registrations, and also some that are missing, and some that are impossible here ("layout:section:%s:collapsed").

Reply:
<Note to self> Revert to old behaviour with new plug name.

Reply:
https://github.com/GafferHQ/gaffer/pull/3491/commits/a85a01c0fbe1881ab2480036a5c217cd0b86782d